Resources/doc/suggesters/indexed_suggesters.rst

```diff
@@ -0,0 +1,231 @@
+Indexed suggesters
+==================
+
+Indexed suggesters require special mapping and documented data to work,
+this includes two suggesters: completion and context.
+
+Mapping
+-------
+
+Completion
+~~~~~~~~~~
+
+Starting with a more simple example, completion suggester. Completion suggester is a simple field which you would just define in a document or an object using a special annotation for suggesters. Available mapping parameters can be found `here`_. For example:
+
+.. code:: php
+
+    <?php
+
+    namespace Acme\DemoBundle\Document;
+
+    use ONGR\ElasticsearchBundle\Annotation as ES;
+
+    /**
+     * Product document.
+     *
+     * @ES\Document(type="product")
+     */
+    class Product implements DocumentInterface
+    {
+        /**
+         * @var CompletionSuggesting
+         *
+         * @ES\Suggester\CompletionSuggesterProperty(
+         *  name = "completion_suggesting",
+         *  objectName = "AcmeDemoBundle:CompletionSuggesting",
+         *  index_analyzer = "simple",
+         *  search_analyzer = "simple",
+         *  payloads = false,
+         *  )
+         */
+        public $completionSuggesting;
+    }
+
+As you can see we need to define an object for it, so we can store and retrieve its data objectively.
+In order to do that, we can just use a trait already prepared in ESB.
+
+.. code:: php
+
+    <?php
+
+    namespace Acme\DemoBundle\Document;
+
+    use ONGR\ElasticsearchBundle\Annotation as ES;
+    use ONGR\ElasticsearchBundle\Document\Suggester\CompletionSuggesterInterface;
+    use ONGR\ElasticsearchBundle\Document\Suggester\CompletionSuggesterTrait;
+
+    /**
+     * Suggesting document for testing.
+     *
+     * @ES\Object
+     */
+    class CompletionSuggesting implements CompletionSuggesterInterface
+    {
+        use CompletionSuggesterTrait;
+    }
+
+Context
+~~~~~~~
+
+Context suggester does not only use the `parameters`_ used in completion suggester, but you must also set additional context mapping, into which you’ll store your data. To do this, ESB uses special annotation objects.
```

Comment:
Context suggester not only uses the `parameters`_ used in completion suggester, but also additional context mapping, into which you’ll store your data.